### ISSUE TITLE
Lint/UselessMethodDefinition: false positive(?) with mixed in method

### ISSUE THREAD

**Issue by leoarnold** [rubocop/rubocop#8678]
## Expected behavior

Given the test script

```ruby
require 'action_controller'

class A
  include ActiveSupport::Configurable
  include ActionController::RequestForgeryProtection

  def valid_authenticity_token?(session, encoded_masked_token)
    super
  end
end

class B
  include ActiveSupport::Configurable
  include ActionController::RequestForgeryProtection
end
```

the method definition in class `A` seems necessary

```ruby
$ irb -I. -rtest.rb
2.5.8 :001 > A.new.respond_to?(:valid_authenticity_token?)
 => true 
2.5.8 :002 > B.new.respond_to?(:valid_authenticity_token?)
 => false 
```

and therefore is not an offense.

## Actual behavior

```ruby
$ rubocop --only Lint/UselessMethodDefinition test.rb 
Inspecting 1 file
W

Offenses:

test.rb:7:3: W: Lint/UselessMethodDefinition: Useless method definition detected.
  def valid_authenticity_token?(session, encoded_masked_token) ...
  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense auto-correctable
```

## Steps to reproduce the problem

See above

## RuboCop version

```
$ [bundle exec] rubocop -V
0.90.0 (using Parser 2.7.1.4, rubocop-ast 0.3.0, running on ruby 2.5.8 x86_64-linux)
```

**Comment by marcandre** [comment]
You are correct it isn't *useless* in the sense of having no side-effect, so maybe we should change the wording, but the only thing this code does is make the method public (and slow it down a bit).

It should be replaced by `public :valid_authenticity_token?`

**Comment by bbatsov** [comment]
Closing as stale — this false positive with mixin method overrides is inherently hard to detect statically and has seen no movement in 5+ years.
